### PR TITLE
Use AdminListComponent for Schemas and Microschemas

### PR DESCRIPTION
**Pull request by lexikon93**

### CODE REVIEW

File: e2e/tests/node-microschema.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    it('should show a right breadcrumb after clicking on "create microschema" and going back to a microschema list', async () => {
+        const breadcrumb: Array<String> = ['Microschemas'];
+
+        await microschema.createNewMicroschemaClick();
+        await microschema.openMicroschema();
+
+        await expect(microschema.getBreadcrumbs()).toEqual(breadcrumb);
+    });
+
+    fit('should create a new schema and delete it (using top button) with selected items check', async () => {
```
Comment (philippguertler) [inline e2e/tests/node-microschema.e2e-spec.ts]: Remove fit

File: e2e/tests/node-microschema.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        await microschema.openMicroschema();
+
+        await microschema.setFilterText('test4');
+
+        await expect(microschema.getAllMicroschemasName()).toEqual(microschemaName);
+
+        await microschema.chooseMicroschema();
+
+        await microschema.clickDeleteTopButton();
+        await microschema.clickDeleteButton();
```
Comment (philippguertler) [inline e2e/tests/node-microschema.e2e-spec.ts]: Rename this function to `clickDialogDeleteButton`

File: e2e/tests/node-microschema.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        await microschema.clickSomeMicroschema('test4');
+        await microschema.clickDeleteBottomButton();
+
+        await microschema.openMicroschema();
+        await microschema.setFilterText('test4');
+
+        await expect(microschema.getAllMicroschemasName()).toEqual(test);
+    });
+
+    it('should goes to the next page and show right microschema', async () => {
```
Comment (philippguertler) [inline e2e/tests/node-microschema.e2e-spec.ts]: should go

File: e2e/tests/node-schema.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        schema = new NodeSchema();
+        await page.navigateToHome();
+        await admin.modeChange();
+        await schema.openSchema();
+    });
+
+    it('should equal schema names', async () => {
+        const list: Array<String> = ['category', 'vehicleImage', 'vehicle', 'binary_content', 'folder'];
+
+        await expect(schema.getAllSchemasName()).toEqual(list);
```
Comment (philippguertler) [inline e2e/tests/node-schema.e2e-spec.ts]: Same as microschema (create schemas by api, etc.)

File: protractor.conf.js [hunk 5]
```diff
@@ -5,9 +5,17 @@ const { SpecReporter } = require('jasmine-spec-reporter');
 
 exports.config = {
   allScriptsTimeout: 11000,
-  specs: [
-    './e2e/**/*.e2e-spec.ts'
-  ],
+  //specs: [
+  //  './e2e/**/*.e2e-spec.ts'
+  //],
```
Comment (philippguertler) [inline protractor.conf.js]: Remove this comment

File: src/app/admin/components/admin-list-item/admin-list-item.component.mock.ts [hunk 6]
```diff
@@ -1,9 +1,11 @@
-import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
 
 @Component({
     selector: 'mesh-admin-list-item',
     template: ``
 })
+
+// tslint:disable-next-line:component-class-suffix
```
Comment (philippguertler) [inline src/app/admin/components/admin-list-item/admin-list-item.component.mock.ts]: Remove this comment and rename the class to `MockAdminListItemComponent`

File: src/app/admin/components/microschema-detail/mircoschema-detail.component.ts [hunk 7]
```diff
@@ -87,9 +154,29 @@ export class MicroschemaDetailComponent implements OnInit, OnDestroy {
...
     }
+
+    onAssignmentChange(project: Project, isChecked: boolean) {
+        if (isChecked) {
+            this.microschema$
+                .take(1)
+                .map(microschema =>
+                    this.schemaEffects.assignEntityToProject('microschema', microschema.uuid, project.name)
+                )
+                .subscribe();
```
Comment (philippguertler) [inline src/app/admin/components/microschema-detail/mircoschema-detail.component.ts]: Remove `map` and put the assignment in the `subscribe` method

File: src/app/admin/components/microschema-detail/mircoschema-detail.component.ts [hunk 8]
```diff
@@ -17,9 +29,23 @@ import { MarkerData } from '../monaco-editor/monaco-editor.component';
 export class MicroschemaDetailComponent implements OnInit, OnDestroy {
     // TODO Disable save button when editor is pristine
     // TODO Show message on save when schema has not changed
+    projects: Project[];
+
+    filterTerm: string;
+
+    filterInput = new FormControl('');
```
Comment (philippguertler) [inline src/app/admin/components/microschema-detail/mircoschema-detail.component.ts]: Check if filter is needed. If not, remove it.

Comment (philippguertler) [reply src/app/admin/components/microschema-detail/mircoschema-detail.component.ts]: Or better: add the filter to the admin list

File: src/app/admin/components/microschema-detail/mircoschema-detail.component.ts [hunk 9]
```diff
@@ -56,10 +84,49 @@ export class MicroschemaDetailComponent implements OnInit, OnDestroy {
...
+            });
+        });
     }
 
     ngOnDestroy(): void {
-        this.subscription.unsubscribe();
+        if (this.subscription) {
+            this.subscription.unsubscribe();
+        }
+        this.destroy$.next();
```
Comment (philippguertler) [inline src/app/admin/components/microschema-detail/mircoschema-detail.component.ts]: Check if this is necessary

File: src/app/admin/components/microschema-list/microschema-list.component.html [hunk 10]
```diff
@@ -1,18 +1,41 @@
-<div class="not-found"
-     *ngIf="(microschemas$ | async).length === 0">
-    <div>
-        <icon>sentiment_very_dissatisfied</icon>
+<mesh-admin-list [items]="microschemas$ | async"
+                 [itemsPerPage]="2"
```
Comment (philippguertler) [inline src/app/admin/components/microschema-list/microschema-list.component.html]: Change to 10

File: src/app/admin/components/project-list/project-list.component.spec.ts [hunk 11]
```diff
@@ -98,50 +98,60 @@ describe('ProjectListComponent', () => {
...
+        )
     );
 
-    it(`deletes a project when delete project is clicked`,
-        componentTest(() => ProjectListComponent, fixture => {
-
-            tick();
-            fixture.detectChanges();
+    it(
+        `deletes a project when delete project is clicked`,
```
Comment (philippguertler) [inline src/app/admin/components/project-list/project-list.component.spec.ts]: Revert the new line

File: src/app/state/models/admin-schemas-state.model.ts [hunk 12]
```diff
@@ -10,4 +12,9 @@ export interface AdminSchemasState {
     schemaDetail: string | null;
     microschemaList: string[];
     microschemaDetail: string | null;
+
+    /** LEX */
```
Comment (philippguertler) [inline src/app/state/models/admin-schemas-state.model.ts]: Remove comments

**Comment by philippguertler** [comment]
Also check the project assignments for multiple projects.
Also when opening a schema, a lot of errors appear in the console

File: src/app/admin/components/project-list/project-list.component.spec.ts (hunk 11 above)
Comment (lexikon93) [reply src/app/admin/components/project-list/project-list.component.spec.ts]: Done for project-list, schema-list and microschema-list.

File: src/app/admin/components/microschema-list/microschema-list.component.html (hunk 10 above)
Comment (lexikon93) [reply src/app/admin/components/microschema-list/microschema-list.component.html]: Done for schema-list.component.html and microschema-list.component.html.

File: e2e/page-objects/node-microschema.po.ts [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+    getMicroschemaButton() {
+        return element.all(by.css('.main-menu li a span')).get(4);
+    }
+
+    getAllMicroschemasName() {
+        return element.all(by.css('.item-primary')).getText();
+    }
+
+    async clickSomeMicroschema(microschemaName: string) {
```
[the remark this answers is not in the document]
Comment (lexikon93) [reply e2e/page-objects/node-microschema.po.ts]: Done forr schema and microschema.

File: e2e/page-objects/node-microschema.po.ts [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+        await this.getSomeMicroschema(microschemaName).click();
+        await browser.sleep(1000);
+    }
+
+    getSomeMicroschema(microschemaName: string) {
```
[the remark this answers is not in the document]
Comment (lexikon93) [reply e2e/page-objects/node-microschema.po.ts]: Done for schema and microschema.

File: e2e/page-objects/node-microschema.po.ts [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    async clickAllocationsTab() {
+        await element
+            .all(by.css('.tab-link a'))
+            .last()
+            .click();
+    }
+
+    getProjectsName() {
```
[the remark this answers is not in the document]
Comment (lexikon93) [reply e2e/page-objects/node-microschema.po.ts]: Done for schema and microschema.

File: e2e/page-objects/node-microschema.po.ts [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+
+    async checkAllMicroschemas() {
+        await element(by.css('.select-all')).click();
+    }
+
+    checkedCount() {
+        return element(by.css('.checked-count'));
+    }
+
+    getBreadcrumbs() {
```
[the remark this answers is not in the document]
Comment (lexikon93) [reply e2e/page-objects/node-microschema.po.ts]: This method was removed from schema and microschema page objects and was added to the admin page object.

File: e2e/page-objects/node-microschema.po.ts [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+    async createNewMicroschemaClick() {
+        await this.getCreateMicroschemaButton().click();
+    }
+
+    getCreateMicroschemaButton() {
+        return element(by.cssContainingText('.button-event-wrapper', 'Create Microschema'));
+    }
+
+    async giveMicroschemaInfo(microschemaInfo: String) {
```
[the remark this answers is not in the document]
Comment (lexikon93) [reply e2e/page-objects/node-microschema.po.ts]: Done for schema and microschema.

File: e2e/page-objects/node-microschema.po.ts [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+
+    async clickDeleteButton() {
+        await element(by.cssContainingText('.button-event-wrapper', 'Delete')).click();
+    }
+
+    async clickDeleteBottomButton() {
+        await browser.waitForAngularEnabled(false);
+        await element
+            .all(by.css('.button-event-wrapper'))
+            .get(4)
```
[the remark this answers is not in the document]
Comment (lexikon93) [reply e2e/page-objects/node-microschema.po.ts]: Done for schema and microschema.

File: e2e/page-objects/node-microschema.po.ts [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+            .click();
+    }
+
+    async goToNextPage() {
+        await element
+            .all(by.css('.page-link'))
+            .get(3)
```
[the remark this answers is not in the document]
Comment (lexikon93) [reply e2e/page-objects/node-microschema.po.ts]: Done for schema and microschema.

File: e2e/tests/node-microschema.e2e-spec.ts (hunk 2 above)
Comment (lexikon93) [reply e2e/tests/node-microschema.e2e-spec.ts]: Done for schema and microschema.

File: src/app/admin/components/microschema-detail/mircoschema-detail.component.ts (hunk 7 above)
Comment (lexikon93) [reply src/app/admin/components/microschema-detail/mircoschema-detail.component.ts]: Done for schema and microschema.

File: src/app/admin/components/microschema-detail/mircoschema-detail.component.ts (hunk 9 above)
Comment (lexikon93) [reply src/app/admin/components/microschema-detail/mircoschema-detail.component.ts]: It is not necessary, was removed from:
- mircoschema-detail.component.ts;
- schema-detail.component.ts;
- mircoschema-list.component.ts;
- mircoschema-list.component.ts.

File: src/app/admin/components/microschema-detail/mircoschema-detail.component.ts (hunk 8 above)
Comment (lexikon93) [reply src/app/admin/components/microschema-detail/mircoschema-detail.component.ts]: Filter for Allocations Tab was added to:
- mircoschema-detail.component.ts;
- schema-detail.component.ts.

File: e2e/tests/node-microschema.e2e-spec.ts [hunk 20]
```diff
@@ -0,0 +1,148 @@
+import { AppPage } from '../page-objects/app.po';
+import { NodeAdmin } from '../page-objects/node-admin.po';
+import { NodeMicroschema } from '../page-objects/node-microschema.po';
+
+describe('node microschema', () => {
+    let admin: NodeAdmin;
+    let page: AppPage;
+    let microschema: NodeMicroschema;
+
```
[the remark this answers is not in the document]
Comment (lexikon93) [reply e2e/tests/node-microschema.e2e-spec.ts]: Functions for create/delete a schema were created in api.ts. BeforeAll method to create test schemas and afterAll method to delete test schemas were implemented.

File: e2e/tests/node-schema.e2e-spec.ts (hunk 4 above)
Comment (lexikon93) [reply e2e/tests/node-schema.e2e-spec.ts]: Functions for create/delete a microschema were created in api.ts. BeforeAll method to create test microschemas and afterAll method to delete test microschemas were implemented.

**Pull request merged**